src/algebra/category/Module/split_exact.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2022 Pierre-Alexandre Bazin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pierre-Alexandre Bazin
+-/
+import algebra.category.Module.abelian
+import algebra.homology.short_exact.abelian
+import algebra.category.Module.biproducts
+
+/-!
+# Split short exact sequences of modules
+
+These lemmas provide an isomorphism `A × B ≃ₗ[R] M` from a left split or right split short exact
+sequence `0 ⟶ A ⟶ M ⟶ B ⟶ 0`, with the "split exact" condition being expressed in terms of
+modules and linear maps without using any category theory.
+-/
+
+universes u v
+variables {R : Type u} {A M B : Type v} [ring R] [add_comm_group A] [module R A]
+  [add_comm_group B] [module R B] [add_comm_group M] [module R M]
+open Module
+
+/--The isomorphism `A × B ≃ₗ[R] M` coming from a right split exact sequence `0 ⟶ A ⟶ M ⟶ B ⟶ 0`
+of modules.-/
+noncomputable def lequiv_prod_of_right_split_exact (j : A →ₗ[R] M) (g : M →ₗ[R] B) (f : B →ₗ[R] M)
+  (hj : function.injective j) (exac : j.range = g.ker) (h : g.comp f = linear_map.id) :
+  (A × B) ≃ₗ[R] M :=
```

Comment:
```suggestion
noncomputable def lequiv_prod_of_right_split_exact {j : A →ₗ[R] M} {g : M →ₗ[R] B} {f : B →ₗ[R] M}
  (hj : function.injective j) (exac : j.range = g.ker) (h : g.comp f = linear_map.id) :
  (A × B) ≃ₗ[R] M :=
```
You can even have a `variables` line, define these only once.

Reply:
For the file, if you mean putting these lemmas in `algebra/homology/short_exact/abelian.lean`, I don't think we want this because this would require importing stuff from `algebra/category/Module` in a file containing general category theory lemmas that don't need it

for variables, done

Reply:
I am in general in favor of having more files, but I am not sure here. Maye adding it to `algebra/category/Module`? I don't think the new import will be a problem.

Reply:
Done